modules/test/playwright/tests/portal-workflow-kaleo-designer-web/workflowDefinitions.spec.ts

```diff
@@ -51,3 +63,118 @@ test('can create a workflow definition with a slash in the name tag', async ({
 		workflowDefinitionName
 	);
 });
+
+test('LPD-49034 Custom Workflow Action CX not working when the asset has related contents', async ({
+	apiHelpers,
+	diagramViewPage,
+	journalEditArticlePage,
+	journalPage,
+	page,
+	processBuilderPage,
+	site,
+	workflowPage,
+}) => {
+	const workflowDefinitionName = getRandomString();
+
+	const workflowDefinition =
+		await apiHelpers.headlessAdminWorkflow.postWorkflowDefinitionSave(
+			workflowDefinitionName,
+			getWorkflowDefinition('simple-cx')
+		);
+
+	workflowDefinitionIds.push(workflowDefinition.id);
+
+	await processBuilderPage.goto();
+
+	await processBuilderPage.clickWorkflowDefinitionName(
+		workflowDefinitionName
+	);
+
+	await expect(diagramViewPage.workflowDefinitionTitle).toHaveValue(
+		workflowDefinitionName
+	);
+
+	await diagramViewPage.publishWorkflowDefinition();
+
+	await diagramViewPage.goBack();
+
+	await workflowPage.goto(site.friendlyUrlPath);
+
+	await workflowPage.changeToCustomWorkflow(
+		'Web Content Article',
+		workflowDefinitionName
+	);
+
+	const basicWebContentStructureId =
+		await getBasicWebContentStructureId(apiHelpers);
+
+	const journalArticleTitle1 = getRandomString();
+
+	await apiHelpers.jsonWebServicesJournal.addWebContent({
+		ddmStructureId: basicWebContentStructureId,
+		groupId: site.id,
+		titleMap: {en_US: journalArticleTitle1},
+	});
+
+	await journalPage.goto(site.friendlyUrlPath);
+
+	await expect(page.getByText(`${journalArticleTitle1}`)).toBeVisible();
+
+	await journalPage.goto(site.friendlyUrlPath);
+
+	await expect(
+		page.locator(
+			'[id="_com_liferay_journal_web_portlet_JournalPortlet_articles_1"] span.label-item'
+		)
+	).toHaveText('Approved');
+
+	const journalArticleTitle2 = getRandomString();
+
+	await journalPage.goToCreateArticle();
+
+	await journalPage.fillArticleData(journalArticleTitle2, getRandomString());
+
+	const row = page
+		.frameLocator('iframe[title="Select Basic Web Content"]')
+		.locator('.list-group-item', {hasText: journalArticleTitle1});
+
+	await expect(async () => {
+		await journalEditArticlePage.openRelatedAsset('Basic Web Content');
+
+		await expect(page.getByText('Select Basic Web Content')).toBeVisible({
+			timeout: 3000,
+		});
+	}).toPass();
+
+	await row.getByRole('checkbox').check();
+
+	await clickAndExpectToBeHidden({
+		target: page.locator('.modal-dialog'),
+		trigger: page.getByRole('button', {name: 'Done'}),
+	});
+
+	await journalEditArticlePage.submitArticleForWorkflow(journalArticleTitle2);
+
+	await journalPage.goto(site.friendlyUrlPath);
+
+	await expect(
+		page.locator(
+			'[id="_com_liferay_journal_web_portlet_JournalPortlet_articles_1"] span.label-item'
```

Comment:
If you can find a better locator for the last comment, remember to use it here as well.